### PR TITLE
Draft: [SYCL][CUDA] Move queue object and lifetime entry points to UR

### PR DESCRIPTION
**Pull request by callumfare**
Draft changes to move the queue object + entry points and the event object to UR. Based on top of #8380, which is in turn based on other merge requests.

### CODE REVIEW

**Review by fabiomestre — COMMENTED** [review]
It's really hard to follow all the major changes in this PR. Most of it seems to be moving code from PI and / or renaming things. I found myself trying to compare code between PI and UR to see if anything had changed.

If possible, could you point to areas where you think there was significant changes done. I can do another round of reviewing and pay closer attention to those specific functions / files.

It looks good to me overall though.

**Comment by callumfare** [comment]
Closing this as it is superseded by https://github.com/intel/llvm/pull/9009
